ledger/acctupdates.go

```diff
@@ -292,51 +295,63 @@ func (au *accountUpdates) lookup(rnd basics.Round, addr basics.Address, withRewa
 }
 
 func (au *accountUpdates) listAssets(maxAssetIdx basics.AssetIndex, maxResults uint64) ([]basics.CreatableLocator, error) {
+	return au.listCreatables(basics.CreatableIndex(maxAssetIdx), maxResults, basics.AssetCreatable)
+}
+
+func (au *accountUpdates) listApplications(maxAppIdx basics.AppIndex, maxResults uint64) ([]basics.CreatableLocator, error) {
+	return au.listCreatables(basics.CreatableIndex(maxAppIdx), maxResults, basics.AppCreatable)
+}
+
+func (au *accountUpdates) listCreatables(maxCreatableIdx basics.CreatableIndex, maxResults uint64, ctype basics.CreatableType) ([]basics.CreatableLocator, error) {
 	au.accountsMu.RLock()
 	defer au.accountsMu.RUnlock()
 
-	// Sort indices for assets that have been created/deleted. If this
+	// Sort indices for creatables that have been created/deleted. If this
 	// turns out to be too inefficient, we could keep around a heap of
 	// created/deleted asset indices in memory.
-	keys := make([]basics.AssetIndex, 0, len(au.assets))
-	for aidx := range au.assets {
-		if aidx <= maxAssetIdx {
-			keys = append(keys, aidx)
+	keys := make([]basics.CreatableIndex, 0, len(au.creatables))
+	for cidx, delta := range au.creatables {
+		if delta.ctype != ctype {
+			continue
+		}
+		if cidx <= maxCreatableIdx {
+			keys = append(keys, cidx)
 		}
 	}
 	sort.Slice(keys, func(i, j int) bool { return keys[i] > keys[j] })
 
-	// Check for assets that haven't been synced to disk yet.
-	var unsyncedAssets []basics.CreatableLocator
-	deletedAssets := make(map[basics.AssetIndex]bool)
-	for _, aidx := range keys {
-		delta := au.assets[aidx]
+	// Check for creatables that haven't been synced to disk yet.
+	var unsyncedCreatables []basics.CreatableLocator
+	deletedCreatables := make(map[basics.CreatableIndex]bool)
```

Comment:
Nit: You might be able to get away with having the map value be a struct instead as `map[basics.CreatableIndex]struct{}`.